src/components/TagContainer.ts

```diff
@@ -0,0 +1,277 @@
+            tagList: []
+        };
+
+        this.createTag = this.createTag.bind(this);
+        this.processTags = this.processTags.bind(this);
+        this.lazyLoadSuggestions = this.lazyLoadSuggestions.bind(this);
+        this.removeTag = this.removeTag.bind(this);
+        this.tagEntity = props.tagEntity.split("/")[props.tagEntity.split("/").length - 1];
+        this.referenceAttribute = props.tagEntity.split("/")[0];
+    }
+
+    render() {
+        return createElement(Tag, {
+            alertMessage: this.state.alertMessage,
+            tagStyle: this.props.tagStyle,
+            className: this.props.class,
+            createTag: this.createTag,
+            enableSuggestions: this.props.enableSuggestions,
+            inputPlaceholder: this.props.inputPlaceholder,
+            lazyLoad: this.props.lazyLoad,
+            fetchSuggestions: this.lazyLoadSuggestions,
+            newTag: this.state.newTag,
+            onRemove: this.removeTag,
+            readOnly: this.isReadOnly(),
+            style: ValidateConfigs.parseStyle(this.props.style),
+            suggestions: this.state.suggestions,
+            tagLimit: this.props.tagLimit,
+            tagLimitMessage: this.props.tagLimitMessage,
+            tagList: this.state.tagList
+        });
+    }
+
+    componentWillReceiveProps(newProps: TagContainerProps) {
+        this.fetchTags(newProps.mxObject);
+        this.resetSubscriptions(newProps.mxObject);
+    }
+
+    componentWillUnmount() {
+        this.subscriptionHandles.forEach(mx.data.unsubscribe);
+    }
+
+    private isReadOnly() {
+        const { editable, mxObject, readOnly } = this.props;
+
+        if (editable === "default" && mxObject) {
+            return readOnly;
+        }
+
+        return true;
+    }
+
+    private resetSubscriptions(mxObject?: mendix.lib.MxObject) {
+        this.subscriptionHandles.forEach(window.mx.data.unsubscribe);
+
+        if (mxObject) {
+            this.subscriptionHandles.push(mx.data.subscribe({
+                callback: guid => {
+                    mx.data.get({
+                        guid: guid.toString(),
+                        callback: object => this.fetchTags(object)
+                    });
+                },
+                guid: mxObject.getGuid()
+            }));
+
+            this.subscriptionHandles.push(mx.data.subscribe({
+                attr: this.referenceAttribute,
+                callback: guid => {
+                    mx.data.get({
+                        guid: guid.toString(),
+                        callback: object => this.fetchTags(object)
+                    });
+                },
+                guid: mxObject.getGuid()
+            }));
+
+            this.subscriptionHandles.push(mx.data.subscribe({
+                guid: mxObject.getGuid(),
+                val: true,
+                callback: validations => {
+                    window.mx.ui.error(validations[0].getErrorReason(this.referenceAttribute));
```

Comment:
validations should be handled inline, just like other error messages.
Popups are not user friendly
Popups should be used on direct user feedback error in exceptions like you use it in `fetchTags`